### PR TITLE
fix(specs): ignorePlurals can be a boolean string

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Closes https://github.com/algolia/algoliasearch-client-go/issues/610, turns out boolean string are supported here.
Also add e2e test for `getRule` and `searchRules`, and also for unicode at the same time.

Delete all the `analytics` e2e because the CI won't do it for us.

### CODE REVIEW

File: specs/common/schemas/IndexSettings.yml [hunk 1]
```diff
@@ -1168,3 +1169,9 @@ supportedLanguage:
     - ur
     - uz
     - zh
+
+booleanString:
+  type: string
+  enum:
+    - true
+    - false
```
Comment (shortcuts) [inline specs/common/schemas/IndexSettings.yml:1177]: does it need to be quoted?

File: tests/CTS/requests/search/searchRules.json [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
+            ],
+            "consequence": {
+              "params": {
+                "ignorePlurals": "true"
+              },
+              "filterPromotes": true,
+              "promote": [
+                {
+                  "objectIDs": [
+                    "\u00C6on Flux"
```
Comment (shortcuts) [inline tests/CTS/requests/search/searchRules.json:36]: ahhhhhh

**Review by shortcuts — COMMENTED** [review]
looks great! waiting for the gen

**Review by shortcuts — APPROVED** [review]
GG!
